jobs/vqsr.py

```diff
@@ -0,0 +1,1033 @@
+"""
+Create Hail Batch jobs to create and apply a VQSR models.
+"""
+
+from typing import List, Optional
+import hailtop.batch as hb
+from hailtop.batch.job import Job
+
+from cpg_utils.config import get_config
+from cpg_utils.workflows.utils import can_reuse
+from cpg_utils import Path
+from cpg_utils.hail_batch import reference_path, image_path, command
+from cpg_utils.workflows.resources import STANDARD, HIGHMEM
+
+from jobs.picard import get_intervals
+from jobs.vcf import gather_vcfs
+
+
+# VQSR - when applying model - targets indel_filter_level and snp_filter_level
+# sensitivities. The tool matches them internally to a VQSLOD score cutoff
+# based on the model's estimated sensitivity to a set of true variants.
+SNP_HARD_FILTER_LEVEL = 99.7
+INDEL_HARD_FILTER_LEVEL = 99.0
+
+STANDARD_FEATURES = [
+    'ReadPosRankSum',
+    'MQRankSum',
+    'QD',
+    'FS',
+    'SOR',
+    # 'DP',
+]
+SNP_STANDARD_FEATURES = STANDARD_FEATURES + ['MQ']
+INDEL_STANDARD_FEATURES = STANDARD_FEATURES
+
+ALLELE_SPECIFIC_FEATURES = [
+    'AS_ReadPosRankSum',
+    'AS_MQRankSum',
+    'AS_QD',
+    'AS_FS',
+    'AS_SOR',
+    # Not using depth for the following reasons:
+    # 1. The Broad pipelines don't use it;
+    # 2. -G AS_StandardAnnotation flag to GenotypeGVCFs doesn't include it;
+    # 3. For exomes, depth is an irrelevant feature and should be skipped.
+    # 'AS_VarDP',
+]
+SNP_ALLELE_SPECIFIC_FEATURES = ALLELE_SPECIFIC_FEATURES + ['AS_MQ']
+INDEL_ALLELE_SPECIFIC_FEATURES = ALLELE_SPECIFIC_FEATURES
+
+SNP_RECALIBRATION_TRANCHE_VALUES = [
```

Comment:
Where do these values come from?

Reply:
Here and here
https://github.com/broadinstitute/warp/blob/79261cde9bd06bb6b1d4a83d75dc54f734541fec/pipelines/broad/dna_seq/germline/joint_genotyping/exome/JointGenotyping.inputs.json#L8-L11
https://github.com/broadinstitute/warp/blob/79261cde9bd06bb6b1d4a83d75dc54f734541fec/pipelines/broad/dna_seq/germline/joint_genotyping/wgs/JointGenotyping.inputs.json#L29-L35 (there is no direct example config for WGS AS-VQSR, but adjusted correspondingly)

Reply:
Added a comment into the docstring